packages/server/src/index.ts

```diff
@@ -1090,9 +1143,48 @@ export class App {
             if (filePath.includes('..')) return res.status(500).send(`Invalid file path`)
             //only return from the .flowise openai-assistant folder
             if (!(filePath.includes('.flowise') && filePath.includes('openai-assistant'))) return res.status(500).send(`Invalid file path`)
-            res.setHeader('Content-Disposition', 'attachment; filename=' + path.basename(filePath))
-            const fileStream = fs.createReadStream(filePath)
-            fileStream.pipe(res)
+
+            if (fs.existsSync(filePath)) {
+                res.setHeader('Content-Disposition', contentDisposition(path.basename(filePath)))
+                streamFileToUser(res, filePath)
+            } else {
+                return res.status(404).send(`File ${req.body.fileName} not found`)
+            }
+        })
+
+        this.app.get('/api/v1/get-upload-path', async (req: Request, res: Response) => {
```

Comment:
Reminder to backport this route to the new way modularity express router PR - https://github.com/FlowiseAI/Flowise/pull/1806